### PR TITLE
Hotfix register example

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Fix typo `K8sServiceAccount`
- Migrate register example from flyte-test-manifest to GitHub client 
- Added argument for passing release version in flytectl register
- Now flytectl register example will use GitHub tag as version

```bash
$ flytectl register examples  -d development  -p flytesnacks --k8sServiceAccount=demo
$ flytectl register examples  -d development  -p flytesnacks v0.2.174 
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
| NAME (9)                                                                                                                                                                                              | STATUS  | ADDITIONAL INFO |
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
| /tmp/register516014713/./case_studies/ml_training/house_price_prediction/_pb_output/0_house_price_prediction.house_price_predictor.generate_and_split_data_1.pb                                       | Success | AlreadyExists   |
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
| /tmp/register516014713/./case_studies/ml_training/house_price_prediction/_pb_output/1_house_price_prediction.house_price_predictor.fit_1.pb                                                           | Success | AlreadyExists   |
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
| /tmp/register516014713/./case_studies/ml_training/house_price_prediction/_pb_output/2_house_price_prediction.house_price_predictor.predict_1.pb                                                       | Success | AlreadyExists   |
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
| /tmp/register516014713/./case_studies/ml_training/house_price_prediction/_pb_output/3_house_price_prediction.house_price_predictor.house_price_predictor_trainer_2.pb                                 | Success | AlreadyExists   |
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
| /tmp/register516014713/./case_studies/ml_training/house_price_prediction/_pb_output/4_house_price_prediction.house_price_predictor.house_price_predictor_trainer_3.pb                                 | Success | AlreadyExists   |
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
| /tmp/register516014713/./case_studies/ml_training/house_price_prediction/_pb_output/5_house_price_prediction.multiregion_house_price_predictor.generate_and_split_data_multiloc_1.pb                  | Success | AlreadyExists   |
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
| /tmp/register516014713/./case_studies/ml_training/house_price_prediction/_pb_output/6_house_price_prediction.multiregion_house_price_predictor.parallel_fit_predict_1.pb                              | Success | AlreadyExists   |
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
| /tmp/register516014713/./case_studies/ml_training/house_price_prediction/_pb_output/7_house_price_prediction.multiregion_house_price_predictor.multi_region_house_price_prediction_model_trainer_2.pb | Success | AlreadyExists   |
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
| /tmp/register516014713/./case_studies/ml_training/house_price_prediction/_pb_output/8_house_price_prediction.multiregion_house_price_predictor.multi_region_house_price_prediction_model_trainer_3.pb | Success | AlreadyExists   |
 ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------- -----------------
```

Usages
```
Registers all latest flytesnacks example
::

 bin/flytectl register examples  -d development  -p flytesnacks


Usage

Usage:
  flytectl register examples [flags]

Aliases:
  examples, example, flytesnack, flytesnacks

Flags:
      --archive                       pass in archive file either an http link or local path.
      --assumableIamRole string        custom assumable iam auth role to register launch plans with.
      --continueOnError               continue on error when registering files.
      --dryRun                        execute command without making any modifications.
  -h, --help                          help for examples
      --k8ServiceAccount string        custom kubernetes service account auth role to register launch plans with.
      --outputLocationPrefix string    custom output location prefix for offloaded types (files/schemas).
      --sourceUploadPath string        Location for source code in storage.
      --version string                version of the entity to be registered with flyte. (default "v1")

```
## Type
- [x] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [ ] Code completed
- [ ] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1404
https://github.com/flyteorg/flyte/issues/1410
https://github.com/flyteorg/flyte/issues/1405

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: cmd/config/subcommand/register/files_config.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ type FilesConfig struct {
 	ContinueOnError      bool   `json:"continueOnError" pflag:",continue on error when registering files."`
 	Archive              bool   `json:"archive" pflag:",pass in archive file either an http link or local path."`
 	AssumableIamRole     string `json:"assumableIamRole" pflag:", custom assumable iam auth role to register launch plans with."`
-	K8ServiceAccount     string `json:"k8ServiceAccount" pflag:", custom kubernetes service account auth role to register launch plans with."`
+	K8sServiceAccount    string `json:"k8sServiceAccount" pflag:", custom kubernetes service account auth role to register launch plans with."`
```
[the remark this answers is not in the document]
Comment (evalsocket) [reply cmd/config/subcommand/register/files_config.go:18]: why we need backward compatibility in flytectl ?

File: cmd/register/register_util.go [hunk 2]
```diff
@@ -242,12 +228,12 @@ func hydrateTaskSpec(task *admin.TaskSpec, sourceCode string) error {
 
 func hydrateLaunchPlanSpec(lpSpec *admin.LaunchPlanSpec) {
 	assumableIamRole := len(rconfig.DefaultFilesConfig.AssumableIamRole) > 0
-	k8ServiceAcct := len(rconfig.DefaultFilesConfig.K8ServiceAccount) > 0
+	k8ServiceAcct := len(rconfig.DefaultFilesConfig.K8sServiceAccount) > 0
```
Comment (pmahindrakar-oss) [inline cmd/register/register_util.go:231]: Rename the variable aswell

File: cmd/register/register_util.go [hunk 3]
```diff
@@ -468,32 +454,34 @@ func getJSONSpec(message proto.Message) string {
 	return jsonSpec
 }
 
-func getFlyteTestManifest(org, repository string) ([]FlyteSnack, string, error) {
+func filterExampleFromRelease(releases github.RepositoryRelease) []github.ReleaseAsset {
+	var assets []github.ReleaseAsset
+	for _, v := range releases.Assets {
+		if strings.HasSuffix(*v.Name, ".tgz") {
```
Comment (pmahindrakar-oss) [inline cmd/register/register_util.go:460]: Introduce a function instead which checks whether a file is an archive and put that as a common until function

Comment (evalsocket) [reply cmd/register/register_util.go:460]: No, I think currently it only detects files that are published by us i.e `.tgz` extension, We will ignore the zip extension.

File: cmd/register/examples.go [hunk 4]
```diff
@@ -16,27 +18,41 @@ Registers all latest flytesnacks example
 
  bin/flytectl register examples  -d development  -p flytesnacks
 
+Registers specific release of flytesnacks example
```
Comment (pmahindrakar-oss) [inline cmd/register/examples.go:21]: can we also add a link to github releases to flytectl from which users can get the version

**Review by pmahindrakar-oss — APPROVED** [review]

File: cmd/register/examples.go (hunk 4 above)
Comment (evalsocket) [reply cmd/register/examples.go:23]: Done  
```
{"json":{"src":"examples.go:52"},"level":"info","msg":"Register started for flytesnacks v0.2.179 release https://github.com/flyteorg/flytesnacks/releases/tag/v0.2.179","ts":"2021-09-03T11:45:12+05:30"}

```

**Review by pmahindrakar-oss — APPROVED** [review]
